### PR TITLE
WIP: Fix display of images with non-1 indices

### PR DESCRIPTION
**Pull request by timholy**
Until I finish my fixes to `restrict`, we don't know if there are more changes needed. So don't merge yet.

### CODE REVIEW

**Comment by Evizero** [comment]
This will need a REQUIRE bump for `ImageTransformations 0.2.0` after the merge of https://github.com/JuliaLang/METADATA.jl/pull/8789

(just making a note before I forget about it)

**Comment by timholy** [comment]
I have a more complete PR coming, just waiting for that METADATA PR to merge. I'm not sure the tests cover enough cases of non-1 indices, though, so I'll be curious to see what you think.

**Comment by timholy** [comment]
OK, this sort of works. Two issues:
- are there important cases that need testing but aren't?
- one oddity is that if you run the tests in the REPL, they pass, but if you run the tests a second time in the same session, they don't. That doesn't seem to happen on master.

**Comment by Evizero** [comment]
I'll take a look after a wake-up coffee. Thanks for doing this!

**Comment by Evizero** [comment]
The test oddity that you ran into was a consequence of the last tests overwriting the default color-depth to 24 bit https://github.com/JuliaImages/ImageInTerminal.jl/blob/master/test/tst_baseshow.jl#L20. Thus running the tests again caused the tests to fail that build on the assumption that the default depth is 256.

I pushed a fix to your branch

**Comment by Evizero** [comment]
Concerning tests. I think the current tests look great. I like how you reuse other reference files. This tests output consistency, and keeps the memory footprint a bit in check.

**Comment by Evizero** [comment]
Test checklist for the changed functions:

- [x] `imshow`
- [x] `encodeimg(::SmallBlocks, ::TermColorDepth, ::Matrix)` (lighthouse)
- [x] `encodeimg(::BigBlocks, ::TermColorDepth, ::Matrix)` (lena)
- [x] `encodeimg(::SmallBlocks, ::TermColorDepth, ::Vector)`
- [x] `encodeimg(::BigBlocks, ::TermColorDepth, ::Vector)`

To cover the two remaining `Vector` tests, an adaption of those two testsets should suffice.

- https://github.com/JuliaImages/ImageInTerminal.jl/blob/master/test/tst_encodeimg.jl#L235-L242 
- https://github.com/JuliaImages/ImageInTerminal.jl/blob/master/test/tst_encodeimg.jl#L276-L289

**Comment by Evizero** [comment]
A quick `Vector` test in the REPL revealed the following remaining `OffsetArray` obstacle in `ImageTransformations`: https://github.com/JuliaImages/ImageTransformations.jl/blob/master/src/resizing.jl#L180

**Comment by Evizero** [comment]
If you are too busy for these additional changes in this PR, let me know. I can't implement them right this instance, but I'd be happy to take care of them soonish.

**Comment by timholy** [comment]
An error I first saw yesterday with ImageCore cropped up on AppVeyor. I don't know why that happened, but hopefully the fix is easy.

File: REQUIRE [hunk 1]
```diff
@@ -2,5 +2,5 @@ julia 0.5
 Crayons 0.1.0
 ColorTypes 0.3.2
 ImageCore 0.1.1
-ImageTransformations 0.1.0
+ImageTransformations 0.2.0
```
Comment (Evizero) [inline REQUIRE]: needs a bump to `0.2.1`

**Comment by Evizero** [comment]
Very cool. I am excited about this change!

**Comment by Evizero** [comment]
lgtm. go for merge after CI passes?

**Comment by Evizero** [comment]
Thanks! As primarily a terminal user this change will make hacking on the `WarpedView` a lot nicer

**Pull request merged**